packages/api/src/beacon/routes/beacon/block.ts

```diff
@@ -330,16 +339,29 @@ export function getReqSerializers(config: ChainForkConfig): ReqSerializers<Api,
       schema: {query: {slot: Schema.Uint, parent_root: Schema.String}},
     },
     getBlockRoot: blockIdOnlyReq,
-    publishBlock: reqOnlyBody(AllForksSignedBlockOrContents, Schema.Object),
+    publishBlock: reqOnlyBody(createAllForksSignedBlockOrContents(), Schema.Object),
     publishBlockV2: {
-      writeReq: (item, {broadcastValidation} = {}) => ({
-        body: AllForksSignedBlockOrContents.toJson(item),
-        query: {broadcast_validation: broadcastValidation},
-      }),
-      parseReq: ({body, query}) => [
-        AllForksSignedBlockOrContents.fromJson(body),
-        {broadcastValidation: query.broadcast_validation as BroadcastValidation},
-      ],
+      writeReq: (item, {broadcastValidation, version} = {}) => {
+        const req = {
+          body: createAllForksSignedBlockOrContents(version).toJson(item),
+          query: {broadcast_validation: broadcastValidation},
+        };
+        const headers = version !== undefined ? {"eth-consensus-version": version} : {};
```

Comment:
Version header must be set as per spec, see how this is handled in the refactored api client

https://github.com/ChainSafe/lodestar/blob/c3afb0beaaa5ba9ced4c2c6c68202fecedb18ef3/packages/api/src/beacon/routes/beacon/block.ts#L357-L370